src/v/bytes/tests/BUILD

```diff
@@ -76,3 +76,18 @@ redpanda_cc_btest(
         "@seastar//:testing",
     ],
 )
+
+redpanda_cc_fuzz_test(
+    name = "iobuf_fuzz",
+    timeout = "short",
+    srcs = ["iobuf_fuzz.cc"],
+    args = [
+        "-max_total_time=30",
+        "-rss_limit_mb=8192",
+    ],
+    deps = [
+        "//src/v/base",
+        "//src/v/bytes:iobuf",
+        "//src/v/bytes:scattered_message",
+    ],
+)
```

Comment:
> We don't limit to -O0

The release profile also doesn't include debug info. With optimizations and no symbolic traces, presumably any errors surfaced by a would be pretty hard to make sense of.

Is ubsan likely to catch errors in an optimized build that was missed in an unoptimized one? I'd sort of expect the result of a full optimization pass to be _less_ recognizable to the tooling, but idk.

Reply:
> With optimizations ~and no symbolic traces~, presumably any errors would be pretty hard to make sense of.

Probably you are right, but it would be better to know that there is something potentially wrong. It might lead to painful debugging but i would argue it is better than not knowing at all. :man_shrugging:

Reply:
> I'd sort of expect the result of a full optimization pass to be less recognizable to the tooling, but idk.

honestly, i don't know either.
My only counter-arguments are "we should be testing the code that we release" and
the quote

> It might be interesting to run w.g. with -O1 and -O2 due to different inlining. 
Running -O0 is rarely beneficial and is very slow. 

From this question
https://groups.google.com/g/libfuzzer/c/L-RQkOaJpoQ

Reply:
Yeah makes sense. Particularly the bit about inlining. I wonder why we only build these in Debug in vtools... maybe @dotnwat can comment?

Reply:
>TODO: figure out why the fuzz test is failing in the ci after the fuzzer has finished testing. It appears to be some hiccup while seastar is doing cleanup, but i need to look further into it.

I tried on a dummy fuzz test that does nothing. Just by linking against seastar, this error occurs. @dotnwat , could it be this was the reason it was set as debug-only?

Reply:
> I wonder why we only build these in Debug in vtools... maybe @dotnwat can comment?

The reason fuzz test is only built in debug is because that is the only build that contains sanitizers. If you build the fuzz tests with release, then you only get feedback on things like segfault.